src/orion/core/worker/trial.py

```diff
@@ -236,6 +238,27 @@ def is_registered(self):
         """Check whether `Trial` is registered in database based on `_id` value."""
         return self._id is not self.NoID
 
+    def params_repr(self, sep=','):
+        """Represent with a string the parameter contained in this `Trial` object."""
+        return sep.join(map(lambda param: "{0.name}:{0.value}".format(param), self.params))
+
+    @property
+    def hash_name(self):
+        """Generate a unique name with an md5sum hash for this `Trial`.
+
+        .. note:: Two trials that have the same `params` must have the same `hash_name`.
```

Comment:
I disagree. Two trials with identical parameters could have different results. We can't assume the user's script is deterministic. We could, however, add a `seed` to the attributes of the Trial and pass it through an environment variable to the user's script. The user could seed the script using it, making it deterministic for a given pair of (params, seed).

Reply:
Is it the case that we may request from the script to evaluate a particular set of parameters twice? Then, that's a design issue that we did not take care about.

only if the `params` are the same, `hash_name` is going to be, not the `results`.

#71 is this the one that you opened because of this observation?

Reply:
Yes I opened #71 for this.

If two trials with identical parameters give different results then they should be considered different, hence have different hash names.

For hyper-parameter optimization, it would not make sense to run twice the exact same hyper-parameters unless they are all categoricals or very small interval of integers. If any is dimension is real or large interval of integers, than the optimization algorithm is better to poke around than running twice identical sets.

However, someone could want to run multiple trials with identical params to compute the variance of the performance. This is something we could support.

To make it clear, what I propose is to keep it like you did for now, our first algorithms won't use seed information anyway. But I believe we should add support for seeds later on, and include the seed information in what we use to compute the hash name of a trial. A change like that should not break anything so it's ok to postpone it. (it's possible to keep backward compatibility for prior trials after we add support for seeds)

Reply:
`hash_name` is supposed to be generated and given as an input to user's script so that it is able to disambiguate local directories that it may use (i.e. for snapshots or logs) among different hyperparameter trial sets. It is supposed to be generated by trials whose results are not yet known, as is the use case above. So this hash_name corresponds only to `trial.params` only. To make it able to run multiple experiments for the same set of hyperparameters:
1. yes we should provide the seed to the user's script. I predict a `~trial.seed` kind of command in the likes of what I've built here. So, `seed` is going to be an attribute of a `Trial` as well.
2. `hash_name` will be generated as a function of `(trial.params, trial.seed)` to ensure uniqueness of results and reproducibility.